### PR TITLE
Add script to generate multiple test cases

### PR DESCRIPTION
**Pull request by jkozol**
This script generates json test cases for each supported output format. It requires an architecure, distro, osbuild store, and output directory as input arguments. It formats and passes this info to the generate-test-case (previously generate-test-cases) script which creates the individual test cases. Then this script writes these json test cases to files in the specified output directory.

### CODE REVIEW

File: tools/generate-test-case [hunk 1]
```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/python3
```
Comment (teg) [inline tools/generate-test-case]: I think you did a `git add` rather than `git mv`, which makes both the patches a bit hard to follow. If you first do the rename, then do a second patch to add a boot type to this script (or the other way around), and then add the new script in the final patch, I think it will be clearer.
Very happy about the actual change here though. Love the in-line documentation.

File: tools/generate-test-cases [hunk 2]
```diff
@@ -3,84 +3,75 @@
...
-    return test_case
 
+def main(distro, arch, store, output):
+    test_case = []
+    for output_format, boot in format_boot_dict.items():
+        test_case_request = {
+            "boot": {
+                "type": boot
+            },
+            "compose-request": {
```
Comment (teg) [inline tools/generate-test-cases]: I don't think this works. i think you need a map from image type to each test_case_request, as the blueprints will be different. AMI uses an empty blueprint, the others have the ssh key, and some have the openssh-server package explicitly enabled. Running the script should not change the actual test cases, but this does, right? (could add some new fields to rpmmd, but that's not crucial as they are unused).
Overall this patch looks great though!

File: tools/generate-test-cases [hunk 3]
```diff
@@ -3,84 +3,75 @@
...
+            json.dump(test_case, case_file, indent=2)
+    return
 
 if __name__ == '__main__':
     parser = argparse.ArgumentParser(description="Generate test cases")
-    parser.add_argument("store", metavar="DIRECTORY", type=os.path.abspath, help="path to the osbuild store")
+    parser.add_argument("--distro", help="distribution for test cases", required=True)
+    parser.add_argument("--arch", help="architecture for test cases", required=True)
+    parser.add_argument("--store", metavar="STORE_DIRECTORY", type=os.path.abspath, help="path to the osbuild store", required=True)
+    parser.add_argument("--output", metavar="OUTPUT_DIRECTORY", type=os.path.abspath, help="path to the osbuild store", required=True)
```
Comment (ondrejbudai) [inline tools/generate-test-cases]: `help` parameter seems to be copy-pasted from the previous line

File: tools/generate-test-case [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+                    subprocess.run(["xz", "--decompress", "--stdout", image_file], stdout=f)
+                image_file = fn
+        test_case["image-info"] = json.loads(subprocess.check_output(["tools/image-info", image_file]))
+
+    return test_case
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(description="Generate test cases")
+    parser.add_argument("store", metavar="DIRECTORY", type=os.path.abspath, help="path to the osbuild store")
```
Comment (ondrejbudai) [inline tools/generate-test-case]: I suggest adding `required=True` for the sake of consistency.

File: tools/generate-test-case (hunk 1 above)
Comment (jkozol) [reply tools/generate-test-case]: Improved the test case

File: tools/generate-test-cases (hunk 2 above)
Comment (jkozol) [reply tools/generate-test-cases]: I now only use a single mapping between the test case request and the output type. The test case requests are located in the format-request-map.json.

File: tools/generate-test-cases (hunk 3 above)
Comment (jkozol) [reply tools/generate-test-cases]: Thanks for pointing this out. I have updated the help text.

File: tools/generate-test-case (hunk 4 above)
Comment (jkozol) [reply tools/generate-test-case]: Since store is a positional argument it is required regardless of having required=True. For consistency, it could be nice to have the store assigned to a flag `--store` as it is in generate-test-cases but since the store is the only argument here I think it is fine as a positional argument.
